proposal/design/zh/metering设计.md

```diff
@@ -0,0 +1,221 @@
+# Metering Proposal
+
+## 一、背景介绍
+
+需要计量计费用户在 sealos cloud 上面使用的资源，比如使用的内存，CPU，流量等资源，每个用户至少有一个自己的 Namespace ，内存和 CPU 等资源 Metering 可以通过直接遍历 pod 得到，但是流量等第三方资源 Metering 感知不到，需要设计一个计量计费系统来帮助计费。
+
+整体流程会在官网的 design 文档中介绍， proposal 中会侧重于介绍 CRD 字段设计细节。
+
+## 二**、各个模块介绍**
+
+### 1**、Metering（计量计费模块合一）**
+
+计费方式：used/unit *price 就是需要扣除的的价格
+
+```go
+apiVersion: metering.sealos.io/v1
+kind: Metering
+metadata:
+  name: metering-nsName
+  namespace:metering-system
+spec:
+  namespace: ns-ff839a27-0a35-452f-820e-3e47d596ba68
+  owner: ff839a27-0a35-452f-820e-3e47d596ba68
+  resources:
+    cpu:
+      describe: cost per cpu per hour（price:100 = 1¥）
+      price: 67
+      unit: "10"
+      used: "0"
+    memory:
+      describe: the cost per gigabyte of memory per hour（price:100 = 1¥）
+      price: 33
+      unit: 10G
+      used: "0"
+  timeInterval: 60 //计费60分钟计费一次
+```
+
+### 2、Resource-controller（以 podResource-controller举例）
+
+进行 resource 的统计，会创建 extensionresourceprice 来声明 cpu 和 memory 这两种资源要进行计量计费，并且将资源的申明注入到 Metering 中。
+
+```yaml
+apiVersion: metering.sealos.io/v1
+kind: PodResource
+metadata:
+  name: podresourceprice-sample
+  namespace: metering-system
+spec:
+  resourceName: pod
+  interval: 60
+  resources:
+    cpu:
+      unit: "10"
+      price: 67
+      describe: "cost per cpu per hour（price:100 = 1¥）"
+    memory:
+      unit: "10G"
+      price: 33
+      describe: "the cost per gigabyte of memory per hour（price:100 = 1¥）"
+```
+
+### 3、**ExtensionResourcesPrice**
+
+由 resource-controller 创建，创建的时候会把资源和价格更新到对应的 Metering
+
+```yaml
+apiVersion: metering.sealos.io/v1
+Kind: ExtensionResourcesPrice
+Spec:
+resources: 
+  - name:cpu
+  unit: 10  //单位使用资源
```

Comment:
missing space for `unit` and `price`, all three with `name` are one list of resource.